src/app/routes/admin-forms.server.routes.js

```diff
@@ -190,7 +199,89 @@ module.exports = function (app) {
   app
     .route('/:formId([a-fA-F0-9]{24})/adminform')
     .get(withUserAuthentication, AdminFormController.handleGetAdminForm)
-    .put(authActiveForm(PermissionLevel.Write), adminForms.update)
+    .put(
+      withUserAuthentication,
+      celebrate({
+        [Segments.BODY]: {
+          form: Joi.object({
+            editFormField: Joi.object({
+              action: {
+                name: Joi.string().valid(...Object.values(EditFieldActions)),
+                position: Joi.when('form.editFormField.action.name', {
+                  is: EditFieldActions.Reorder,
+                  then: Joi.number().required(),
+                }),
+              },
+              field: Joi.object({
+                fieldType: Joi.string()
```

Comment:
I'm wondering whether it's worth extracting all these specifications so they can be reused across multiple endpoints in a modular fashion. eg this `fieldType` key is repeated in public submissions, preview submissions and here.

Reply:
Can probably do this when everything is in Typescript